pulp_ansible/tests/functional/api/collection/v3/test_collection.py

```diff
@@ -248,6 +249,51 @@ def test_collection_version_list(artifact, pulp_client, collection_detail, colle
     assert version["href"] == collection_detail["highest_version"]["href"]
 
 
+def test_collection_version_filter_by_q(pulp_client, pulp_dist):
+    """Verify successive imports do not aggregate tags into search vectors."""
+
+    def publish(new_artifact):
```

Comment:
I think this test can be re-written to use the client bindings fixture `ansible_collection_version_api_client`. It should make the test easier to read and will correctly work even with rerouted tests like our S3 runner.

```python
def publish(new_artifact):
    body = {
        "file": os.path.basename(new_artifact.filename),
        "namespace": new_artifact.namespace,
        "name": new_artifact.name, 
        "version": new_artifact.version
        }
    resp = ansible_collection_version_api_client.create(**body)
    monitor_task(resp.task)
```

```python
for spec in specs:
    resp = ansible_collection_version_api_client.list(q=spec[0])
```